File: py/bank.py
from money import Money


class Bank:
    def __init__(self) -> None:
        self.exchangeRates = {}

    def addExchangeRate(self, currencyFrom: str, currencyTo: str, rate: float) -> None:
        key = currencyFrom + '->' + currencyTo
        self.exchangeRates[key] = rate

    def convert(self, aMoney: Money, aCurrency: str):
        if aMoney.currency == aCurrency:
            return Money(aMoney.amount, aCurrency)
        key = aMoney.currency + '->' + aCurrency
        if key in self.exchangeRates:
            return Money(aMoney.amount * self.exchangeRates[key], aCurrency)
        raise Exception(key)

File: js/bank.js
const Money = require("./money");

class Bank {
  constructor() {
    this.exchangeRates = {};
  }

  addExchangeRate(currencyFrom, currencyTo, rate) {
    const key = currencyFrom + "->" + currencyTo;
    this.exchangeRates = { ...this.exchangeRates, [key]: rate };
  }

  convert(money, currency) {
    if (money.currency == currency) {
      return new Money(money.amount, money, currency);
    }
    const key = money.currency + "->" + currency;

    const rate = this.exchangeRates[key];
    if (!rate) {
      throw new Error(key);
    }

    return new Money(money.amount * rate, currency);
  }
}

module.exports = Bank;
